extension-manifest-v3/src/background/reporting.js

```diff
@@ -267,6 +286,16 @@ function onLocationChange(details) {
     }
 
     try {
+      if (
```

Comment:
Like the comment above, this will fail if the state changes.

Dropping the "if" and only awaiting the defer (as if waiting if the initial setup is complete) should work. And let the defer resolve without any value.

Or let it always resolve to "true" to indicate the the initial startup is done (but IMHO that is less clear).